server/app/services/program/ProgramService.java

```diff
@@ -682,11 +702,39 @@ private ImmutableSet<CiviFormError> validateProgramData(
     if (displayName.isBlank()) {
       errorsBuilder.add(CiviFormError.of(MISSING_DISPLAY_NAME_MSG));
     }
-    if (displayDescription.isBlank()) {
+    if (displayDescription.isBlank()) { // remove this check now that this is no longer required
       errorsBuilder.add(CiviFormError.of(MISSING_DISPLAY_DESCRIPTION_MSG));
-    } else if (displayMode.equals(DisplayMode.SELECT_TI.getValue()) && tiGroups.isEmpty()) {
+    } else if (displayMode.equals(DisplayMode.SELECT_TI.getValue())
+        && tiGroups.isEmpty()) { // should this else statement be below on displayMode??
       errorsBuilder.add(CiviFormError.of(MISSING_TI_ORGS_FOR_THE_DISPLAY_MODE));
     }
+    if (shortDescription.isBlank()) {
+      errorsBuilder.add(CiviFormError.of(MISSING_DISPLAY_DESCRIPTION_MSG));
+    }
+    boolean havePreviousStep = false;
+    for (int i = 0; i < applicationSteps.size(); i++) {
```

Comment:
We'll probably want to make steps validation it's own method since this is a lot of logic.